### PR TITLE
FS2-2477: Fix up axis for different axis data types

### PR DESCRIPTION
**Pull request by chschan**

### CODE REVIEW

File: theSrc/test/snapshots/travis/master/testPlans/functionalTest/bounds-9-snap.png
Comment (JustinCCYap) [inline theSrc/test/snapshots/travis/master/testPlans/functionalTest/bounds-9-snap.png:1]: The labels are missing here, I assume it is flakiness? If due to flakiness, can we remove the revert this snapshot change?

Comment (JustinCCYap) [reply theSrc/test/snapshots/travis/master/testPlans/functionalTest/bounds-9-snap.png:1]: Compared to https://github.com/Displayr/rhtmlLabeledScatter/blob/master/theSrc/test/snapshots/travis/master/testPlans/functionalTest/bounds-9-snap.png, it seems we still need to remove the empty legend, and change the range (not as part of this PR, but as extra subtasks)?

File: theSrc/test/snapshots/travis/master/testPlans/functionalTest/prefix_suffix-4-snap.png
Comment (JustinCCYap) [inline theSrc/test/snapshots/travis/master/testPlans/functionalTest/prefix_suffix-4-snap.png:1]: I think labels should be shown here?

File: theSrc/test/snapshots/travis/master/testPlans/functionalTest_old/functionaltest_old_data_functionaltest_scatterplot_y_datetimes-snap.png
Comment (JustinCCYap) [inline theSrc/test/snapshots/travis/master/testPlans/functionalTest_old/functionaltest_old_data_functionaltest_scatterplot_y_datetimes-snap.png:1]: rhtmlLabeledScatter used to provide extra space for large bubbles at the bounds. Could we do something similar with plotly? Here is the snapshot from rhtmlLabeledScatter: https://github.com/Displayr/rhtmlLabeledScatter/blob/master/theSrc/test/snapshots/travis/master/testPlans/functionalTest_old/functionaltest_old_data_functionaltest_scatterplot_y_datetimes-snap.png

**Review by JustinCCYap — APPROVED** [review]

File: theSrc/test/snapshots/travis/master/testPlans/functionalTest_old/functionaltest_old_data_functionaltest_scatterplot_y_datetimes-snap.png
Comment (chschan) [reply theSrc/test/snapshots/travis/master/testPlans/functionalTest_old/functionaltest_old_data_functionaltest_scatterplot_y_datetimes-snap.png:1]: Added some extra space for bubbles. Its approximate, but if we notice the approximation behaving badly we could apply it as a call to `Plotly.relayout` after it plots the first time. For now though, this seems good enough and is a bit more efficient

File: theSrc/test/snapshots/travis/master/testPlans/functionalTest/bounds-9-snap.png
Comment (chschan) [reply theSrc/test/snapshots/travis/master/testPlans/functionalTest/bounds-9-snap.png:1]: Fixed up the empty legends in [a202c77](https://github.com/Displayr/rhtmlCombinedScatter/pull/11/commits/a202c77ca4a77902945e04438da02ded8fe86e8b)

**Pull request merged**